src/java.base/share/classes/java/io/File.java

```diff
@@ -245,6 +245,13 @@ int getPrefixLength() {
      */
     public static final String pathSeparator = String.valueOf(pathSeparatorChar);
 
+    /**
+     * The {@code File} object representing the current directory.
+     */
+    private static final File CWD = new File(".");
+    static File getCWD() {
+        return CWD;
+    }
```

Comment:
This is a bit icky, have you looked at pushing this down to a holder class in UnixFileSystem and WinNTFileSystem so they they aren't dependent on this.

Reply:
Why not just down into `FileSystem`? It's not OS-dependent.

Reply:
That would be okay too, just need to avoid circular initialization.

Reply:
So changed in a3b7977.

Reply:
Thanks, that is better replace for this. In time it may be a good candidate to be a stable value.

Reply:
It was a bad idea in the first place that I put that in `File` itself. I also thought of this being a stable value.